pkgs/dart_tooling_mcp_server/test/test_harness.dart

```diff
@@ -122,18 +122,88 @@ class TestHarness {
   }
 }
 
+/// The debug session for a single app.
+///
+/// Should be started using [TestHarness.startDebugSession].
+final class AppDebugSession {
+  final TestProcess appProcess;
+  final String appPath;
+  final String projectRoot;
+  final String vmServiceUri;
+  final bool isFlutter;
+
+  AppDebugSession._(
+      {required this.appProcess,
+      required this.vmServiceUri,
+      required this.projectRoot,
+      required this.appPath,
+      required this.isFlutter});
+
+  static Future<AppDebugSession> _start(String projectRoot, String appPath,
+      {required bool isFlutter}) async {
+    final platform = Platform.isLinux
+        ? 'linux'
+        : Platform.isMacOS
+            ? 'macos'
+            : throw StateError(
+                'unsupported platform, only mac and linux are supported',
+              );
+    final process = await TestProcess.start(
+      isFlutter ? 'flutter' : 'dart',
+      [
+        'run',
```

Comment:
this will still start a second DTD instance in the case of a flutter or dart run:

Flutter Tools --> DevTools Server --> DTD
Dart CLI --> DDS --> DTD

Reply:
Right, but we are just ignoring it right now. My assumption is the IDE also does exactly this and just ignores the DTD instance that comes from flutter/dart run.

Reply:
The IDEs start Dart or Flutter apps and pass the `--no-serve-devtools` (or `--no-dds`?) which results in DTD not being started. 
https://github.com/dart-lang/sdk/blob/main/pkg/dds/lib/src/dds_impl.dart#L197

We could take the same approach here by passing the right flag but we'd need to check what the IDEs are doing currently.